### ISSUE TITLE
Unable to delete temporary data

### ISSUE THREAD

**Issue by RaffaMoreira777** [mikeyy/nonoCAPTCHA#16]
I'm using Windows 7 64 bits, and when I run the script, it works, but it doesn't work after downloading the audio. I receive this error "OSError: Unable to remove Temporary User Data" and the chromium crashs, and when I try to connect to gmail account with a proxy I receive this one "Page.waitForNavigation' was never awaited".
Python Version: 3.6.5
Chromium Auto Installed by pyppeteer Version: 67.0.3372.0

**Comment by mikeyy** [comment]
Thanks for bringing to this to my attention. I will try and push a patch today for both of these issues, bare with me for the next patch update.

**Comment by RaffaMoreira777** [comment]
All right.

**Comment by mikeyy** [comment]
Update released. Give that a try and let me know if you have any issues with it.

**Comment by mikeyy** [comment]
I'm having issues with it still on my end. Will release another one on update 1.4.13, later.

**Comment by RaffaMoreira777** [comment]
```bash
Traceback (most recent call last):
  File "run.py", line 140, in <module>
    loop.run_until_complete(main())
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\asyncio\base_events.py", line 467, in run_until_complete
    return future.result()
  File "run.py", line 120, in main
    result = task.result()
  File "run.py", line 99, in work
    answer = await client.start()
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\site-packages\nonocaptcha\solver.py", line 77, in start
    await self.browser.close()
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\site-packages\pyppeteer\browser.py", line 155, in close
    await self._closeCallback()  # Launcher.killChrome()
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\site-packages\nonocaptcha\launcher.py", line 162, in killChrome
    self._cleanup_tmp_user_data_dir()
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\site-packages\pyppeteer\launcher.py", line 130, in _cleanup_tmp_user_data_dir
    raise IOError('Unable to remove Temporary User Data')
*OSError: Unable to remove Temporary User Data*
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\site-packages\nonocaptcha\launcher.py", line 162, in killChrome
    self._cleanup_tmp_user_data_dir()
  File "C:\Users\User\AppData\Local\Programs\Python\Python36\lib\site-packages\pyppeteer\launcher.py", line 130, in _cleanup_tmp_user_data_dir
    raise IOError('Unable to remove Temporary User Data')
OSError: Unable to remove Temporary User Data
```
I keep with this error. I think my OS doesn't have permission to remove the data of the dev-profiles folder.

**Comment by mikeyy** [comment]
I will look into this further. The issue was arising on my system due to the main Chome process stil running and not allowing the deletion of BrowserMetrics-active.pma in the folder. Can you verify this was tested with version 1.4.13 by typing this in command prompt? 

`python -c "import nonocaptcha; print(nonocaptcha.__version__)"`

If the version differs from 1.4.13 then update nonocaptcha with:

`pip install --upgrade nonocaptcha`

**Comment by RaffaMoreira777** [comment]
I'm using this last version: ``1.4.13``.
The errors persists, even with main chrome closed.

**Comment by mikeyy** [comment]
> This OS error didn't appear this time, I noticed that I wasn't with this last version, so I upgraded to it.

Awesome! Do let me know if it does happen again, I'm not quite sure if it has by your last sentence.

> I notice when the OS error happens, the chrome.exe on process keeps running and the code crashs.

**Comment by RaffaMoreira777** [comment]
It keeps happening but with less frequency, like when I'm using all threads and some of them fail.
At least I found another error while checking google blacklist, I'll make another issue

**Comment by mikeyy** [comment]
Yeah, running multiple threads seems to be giving me problems as well, even on *nix - although it's not the removing directory issue, something keeps hanging and causing the whole process to tumble. I've been at for a couple days trying to find the bug, and it's quite a hard one since it's buried deep inside of Pyppeteer.

Will release an update here in a bit and see if it helps any.

**Comment by mikeyy** [comment]
Just pushed out update 1.4.15, give that one a try. Thanks.

**Comment by mikeyy** [comment]
This issue appears on my end as well but only when using a proxy, which is odd. I'm looking into it, this one might take me a bit longer to fix. Bare with me.

**Comment by mikeyy** [comment]
1.14.16 released, give it a shot and let me know how it goes. I think I came across one other issue but I've already been at the others for 5 hours, _bangs head_

**Comment by RaffaMoreira777** [comment]
I'll test in some minutes. I'm busy atm

**Comment by RaffaMoreira777** [comment]
Checked and It's working perfectly without any crashs. I didn't checked if the audio transport still bugged because my proxies are dead. RIP. I'll try this part soon

**Comment by mikeyy** [comment]
Awesome! Glad it's starting to work on Windows without hiccups. I'm going to close this, don't hesitate to open another issue if a problem pops up.

**Comment by guruprasaad123** [comment]
how do i install 1.14.16 ?

**Comment by mikeyy** [comment]
Latest released version is v2.0.1. It can be installed using `pip`.

**Comment by thewhiterabbit88** [comment]
Hey was having the same issue with being unable to delete temporary files. I down loaded the upgrades and it says that it fail to build multidict. 

---------------
  Building wheel for multidict (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\adam\appdata\local\programs\python\python39\python.exe' 'c:\users\adam\appdata\local\programs\python\python39\lib\site-packages\pip\_vendor\pep517\_in_process.py' build_wheel 'C:\Users\Adam\AppData\Local\Temp\tmp8pui5uqh'
       cwd: C:\Users\Adam\AppData\Local\Temp\pip-install-5a2dkyo8\multidict
  Complete output (40 lines):
  **********************
  * Accellerated build *
  **********************
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.9
  creating build\lib.win-amd64-3.9\multidict
  copying multidict\_abc.py -> build\lib.win-amd64-3.9\multidict
  copying multidict\_compat.py -> build\lib.win-amd64-3.9\multidict
  copying multidict\_multidict_base.py -> build\lib.win-amd64-3.9\multidict
  copying multidict\_multidict_py.py -> build\lib.win-amd64-3.9\multidict
  copying multidict\__init__.py -> build\lib.win-amd64-3.9\multidict
  running egg_info
  writing multidict.egg-info\PKG-INFO
  writing dependency_links to multidict.egg-info\dependency_links.txt
  writing top-level names to multidict.egg-info\top_level.txt
  reading manifest file 'multidict.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no previously-included files matching '*.pyc' found anywhere in distribution
  warning: no previously-included files found matching 'multidict\_multidict.html'
  warning: no previously-included files found matching 'multidict\*.so'
  warning: no previously-included files found matching 'multidict\*.pyd'
  warning: no previously-included files found matching 'multidict\*.pyd'
  no previously-included directories found matching 'docs\_build'
  writing manifest file 'multidict.egg-info\SOURCES.txt'
  copying multidict\__init__.pyi -> build\lib.win-amd64-3.9\multidict
  copying multidict\_multidict.c -> build\lib.win-amd64-3.9\multidict
  copying multidict\py.typed -> build\lib.win-amd64-3.9\multidict
  creating build\lib.win-amd64-3.9\multidict\_multilib
  copying multidict\_multilib\defs.h -> build\lib.win-amd64-3.9\multidict\_multilib
  copying multidict\_multilib\dict.h -> build\lib.win-amd64-3.9\multidict\_multilib
  copying multidict\_multilib\istr.h -> build\lib.win-amd64-3.9\multidict\_multilib
  copying multidict\_multilib\iter.h -> build\lib.win-amd64-3.9\multidict\_multilib
  copying multidict\_multilib\pair_list.h -> build\lib.win-amd64-3.9\multidict\_multilib
  copying multidict\_multilib\views.h -> build\lib.win-amd64-3.9\multidict\_multilib
  running build_ext
  building 'multidict._multidict' extension
  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
  ----------------------------------------
  ERROR: Failed building wheel for multidict
Successfully built aiohttp
Failed to build multidict
ERROR: Could not build wheels for multidict which use PEP 517 and cannot be installed directly
